docs/docs-content/clusters/edge/third-party-integrations/tailscale.md

```diff
@@ -0,0 +1,249 @@
+              cp /oem/tailscale/tailscaled.state /var/lib/tailscale/tailscaled.state
+              systemctl start tailscaled
+              tailscale up --ssh --hostname="edge-${ID}"
+            else
+              tailscale up --authkey=$AUTH-KEY --ssh --hostname="edge-${ID}"
+              mkdir /oem/tailscale
+              cp /var/lib/tailscale/tailscaled.state /oem/tailscale/tailscaled.state
+            fi
+      - name: "Enable Tailscale"
+        if: '[ ! -f "/run/cos/recovery_mode" ] && grep _current-profile /var/lib/tailscale/tailscaled.state'
+        commands:
+          - |
+            ID=$(cat /sys/class/dmi/id/product_uuid)
+            tailscale up --ssh --hostname="edge-${ID}"
+  ```
+
+  If you already have a `stages` block in your user-data file, you must merge the existing block together with the above content. The `stages` block is based on Kairos cloud-init stages. For more information on cloud init stages, refer to [Cloud Init Stages](../edge-configuration/cloud-init.md).
+
+  :::info
+
+  In the above `stages` block, you are using the device ID of your Edge device (read from `/sys/class/dmi/id/product_uuid`) as the hostname with which to register your device with Tailscale. For more information about how this ID is generated, refer to [Install Configurations](../edge-configuration/installer-reference.md#device-id-uid-parameters).
+
+  If you want to use a different hostname, especially when using the `deviceUIDPaths` parameter in the user-data, you can adjust the two `ID=$(cat /sys/class/dmi/id/product_uuid)` lines in the content above to match your custom device naming configuration.
+
+  :::
+
+7. Build the Edge device installation ISO and providers images. Afterward, push the provider images to an image registry. For more information, refer to [Build Edge Artifacts](../edgeforge-workflow/palette-canvos.md).
+
+8. Flash your external volume with the Edge installer ISO image. We recommend using [balena etcher](https://etcher.balena.io/) to flash your volume. 
+
+9. Plug the external volume into your Edge device and boot up the device using the volume to prepare your Edge device for installation. For more information, refer to [Prepare Edge Host for Installation](../site-deployment/stage.md). 
+
+10. Remove the volume and boot up your device again to register your Edge host. If the Edge host has internet access, it will start up Tailscale and register your device with Tailscale. 
+
+
+## Validate
+
+1. Log in to [Tailscale console](https://login.tailscale.com/admin/machines). 
+
+2. In the **Machines** tab, your Edge device is displayed in the Machines list. You can SSH to your host from any device that is also connected to your Tailscale network.
+
+## Troubleshooting
+
+
+### All Traffic Dropped for 100.64.0.0/10 CIDR Range 
+
+Tailscale uses the 100.64.0.0/10 range of IP addresses for your Tailnets. That means that by default, this address range (or parts of it) cannot be used for any of the following:
```

Comment:
```suggestion
Tailscale uses the 100.64.0.0/10 range of IP addresses for your Tailnets. That means that by default, this address range, or parts of it, cannot be used for any of the following:
```
We try to avoid parentheses inside sentences that convey information. The sentences can be rewritten to a more formal tone by omitting the paratheses. Paratheses are also a bit distracting.